### PR TITLE
feat: add css border radius component

### PR DESCRIPTION
**Pull request by eljefe223**
# Description
adds css Border radius component.

image

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.tsx [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+                this.props,
+                "managedClasses.cssBorderRadius_toggleButtonGlyphPath__highlight"
+            );
+        }
+    }
+
+    private handleInputOnFocus(
+        cssKey: BorderRadiusValue
+    ): (e: React.ChangeEvent<HTMLInputElement>) => void {
+        return (e: React.ChangeEvent<HTMLInputElement>): void => {
```
Comment (nicholasrice) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.tsx]: This switch doesn't do anything - you should remove

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+    test("should not throw", () => {
+        expect(() => {
+            shallow(<CSSBorderRadius />);
+        }).not.toThrow();
+    });
+    test("should have a displayName that matches the component name", () => {
+        expect((CSSBorderRadius as any).name).toBe(CSSBorderRadius.displayName);
+    });
+    test("should use the `data` prop as the input value if the `borderRadius` is provided", () => {
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx]: can you add a test to make sure this doesn't throw if `data` is an empty object

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+            },
+        });
+        rendered.setState({ indvidualValues: true });
+
+        const inputTopLeft: any = rendered.find("input").at(0);
+        const inputTopRight: any = rendered.find("input").at(1);
+        const inputBottomRight: any = rendered.find("input").at(2);
+        const inputBottomLeft: any = rendered.find("input").at(3);
+        
+
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx]: nit: extra line

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+            data: {
+                borderRadius: borderRadiusValue
+            },
+        });
+        rendered.setState({ indvidualValues: true });
+
+        const inputTopLeft: any = rendered.find("input").at(0);
+        const inputTopRight: any = rendered.find("input").at(1);
+        const inputBottomRight: any = rendered.find("input").at(2);
+        const inputBottomLeft: any = rendered.find("input").at(3);
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx]: nit: you don't need to store these as values, you can just use find in the expects below

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.props.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import { ManagedClasses } from "@microsoft/fast-jss-manager-react";
+import { CSSBorderRadiusClassNameContract } from "./border-radius.style";
+
+export interface CSSBorderRadiusState {
+    indvidualValues?: boolean;
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.props.ts]: spelling, "individualValues"

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.tsx [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+
+                this.props.onChange(borderRadius);
+        };
+    }
+
+    private toggleInputs = (): void => {
+        this.setState({
+            indvidualValues: !this.state.indvidualValues,
+        });
+
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.tsx]: nit: extra line

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+                onChange={jest.fn()}
+            />
+        );
+
+        rendered.setProps({
+            data: {
+                borderRadius: borderRadiusValue
+            },
+        });
+        rendered.setState({ indvidualValues: true });
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx]: rather than setting state on this, it's preferable to test what action would cause this, so can you use a simulated onClick to the button that toggles this value

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+
+        const inputTopLeft: any = rendered.find("input").at(0);
+        const inputTopRight: any = rendered.find("input").at(1);
+        const inputBottomRight: any = rendered.find("input").at(2);
+        const inputBottomLeft: any = rendered.find("input").at(3);
+        
+
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.spec.tsx]: nit: extra line

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.tsx [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+    constructor(props: CSSBorderRadiusProps) {
+        super(props);
+
+        this.state = {
+            indvidualValues: false,
+            hasFocus: void 0,
+            topLeftValue: "0",
+            topRightValue: "0",
+            bottomRightValue: "0",
+            bottomLeftValue: "0",
```
Comment (janechu) [inline packages/fast-tooling-react/src/css-editor/border-radius/border-radius.tsx]: I would rather we do not store any of the values in state, i wouldn't store data, or individual values. A better pattern would be to have a memoized getter function that when passed in with the border radius could return these values to you. This way you never have to deal with state management and using `componentDidUpdate`

File: packages/fast-tooling-react/src/css-editor/border-radius/border-radius.props.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    topLeftValue?: string;
+    topRightValue?: string;
+    bottomRightValue?: string;
+    bottomLeftValue?: string;
+    data?: CSSBorderRadiusValues;
+}
+
+export enum BorderRadiusValue {
+    borderRadius = "borderRadius",
+    borderTopLeftRadius = "borderTopLeftRadius",
```
[the remark this answers is not in the document]
Comment (eljefe223) [reply packages/fast-tooling-react/src/css-editor/border-radius/border-radius.props.ts]: I don't see anywhere that I'm using them as strings but I get an error if I remove the value assignments:

```
Uncaught TypeError: Failed to set an indexed property on 'CSSStyleDeclaration': Index property setter is not supported.
```

[the remark this answers is not in the document]
Comment (nicholasrice) [reply packages/fast-tooling-react/src/css-editor/border-radius/border-radius.props.ts]: No worries - this is fine.
